include/deal.II/lac/solver_fire.h

```diff
@@ -0,0 +1,392 @@
+#ifndef dealii__solver_fire_h
+#define dealii__solver_fire_h
+
+
+#include <deal.II/base/config.h>
+#include <deal.II/base/logstream.h>
+#include <deal.II/lac/diagonal_matrix.h>
+#include <deal.II/lac/solver.h>
+
+#include <functional>
+
+
+DEAL_II_NAMESPACE_OPEN
+
+
+/*!@addtogroup Solvers */
+/*@{*/
+
+/**
+ * FIRE (Fast Inertial Relaxation Engine) for minimization of (potentially
+ * non-linear) objective function $E(U)$, $U$ is a vector of $n$ variables
+ * ($n$ is the number of variables of the objective function).
+ * Like all other solver classes, it can work on any kind of vector and matrix
+ * as long as they satisfy certain requirements (for the requirements on
+ * matrices and vectors in order to work with this class, see the documentation
+ * of the Solver base class). The type of the solution vector must be passed as
+ * template argument, and defaults to dealii::Vector<double>.
+ *
+ * FIRE is a damped dynamics method described in
+ * <a href="https://doi.org/10.1103/PhysRevLett.97.170201">Structural
+ * Relaxation Made Simple</a> by Bitzek et al. 2006, typically used to find
+ * stable equilibrium configurations of atomistic systems in computational
+ * material science. Starting from a given initial configuration of the
+ * atomistic system, the algorithm relies on inertia to obtain (nearest)
+ * configuration with least potential energy.
+ *
+ * The problem of solving the system of linear equations $AU = F$ for $U$
```

Comment:
i would suggest to use lower-case letters for vectors here and elsewhere. That's what we do elsewhere in documentation, leaving upper case for matrices (be it square matrices for inversions or matrices whose columns are eigenvectors).